packages/flutter/lib/src/material/dialog.dart

```diff
@@ -216,6 +220,19 @@ class AlertDialog extends StatelessWidget {
   /// from the [actions].
   final List<Widget> actions;
 
+  /// An (optional) name for the dialog used by accessibility frameworks.
+  ///
+  /// Defaults to "alert" on both iOS and Android.
+  ///
+  /// This name is announced by TalkBack or VoiceOver when the dialog becomes
+  /// visible. For example, "App permissions" for a dialog containing permission
+  /// information.
+  ///
+  /// See also:
+  ///   * [Semantics.route], the semantics flag which controls screen
+  ///     accessibility.
+  final String routeName;
```

Comment:
AlertDialog has a "title", which we really should just use for the route name. Nobody's ever going to specify the routeName.